include/index_helper.cuh

```diff
@@ -633,8 +636,8 @@ namespace quda {
      @param[in] arg Argument struct with required meta data
      @return Checkerboard lattice index
   */
-  template <int nDim, QudaPCType type, int dim, int nLayers, int face_num, typename Arg>
-  inline __device__ __host__ int indexFromFaceIndex(int face_idx, int parity, const Arg &arg)
+  template <int nDim, typename Arg>
+  inline __device__ __host__ int indexFromFaceIndex(int dim, int face_num, int face_idx, int parity, int nLayers, QudaPCType type, const Arg &arg)
```

Comment:
Can these functions be made `constexpr`?

Reply:
Some of them can easily made `constexpr`, yes.  I have done so in 5e4eb2e8454a34deaea73d0e22aba60fe633b647.